lib/internal/http2/compat.js

```diff
@@ -164,7 +172,16 @@ class Http2ServerRequest extends Readable {
   }
 
   get trailers() {
-    return this[kTrailers];
+    return this[kTrailers] || [];
+  }
+
+  get rawTrailers() {
+    const trailers = this[kTrailers];
+    if (trailers === undefined)
+      return [];
+    const tuples = Object.entries(trailers);
+    const flattened = Array.prototype.concat.apply([], tuples);
```

Comment:
Can you use `[].concat(tuples)` instead? Why is this needed?

Reply:
It's a [[name, value], [name, value]] type of structure. `[].concat(tuples)` won't flatten it.

That said, I would ideally like to land this without this code and instead with a real solution for `rawHeaders` and `rawTrailers` (one that actually provides the real raw data to the user). I'm going to create a new commit with that behaviour and would appreciate if you or @jasnell could review.

Reply:
I think

```js
[].concat(...Object.entries(obj))
```

would read better.